lighthouse-core/test/scripts/i18n/cull-obsolete-lhl-messages-test.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+const cullObsoleteLhlMessages = require('../../../scripts/i18n/cull-obsolete-lhl-messages.js');
+
+/**
+ * Cull `localeLhl` based on `goldenLhl`.
+ * @param {Record<string, {message: string}>} goldenLhl
+ * @param {Record<string, {message: string}>} localeLhl
+ */
+function cullLocale(goldenLhl, localeLhl) {
+  // Suppress logging by adding all message ids to the already-logged set.
+  const loggedCulls = new Set(Object.keys(localeLhl));
+
+  const goldenLocaleArgumentIds = cullObsoleteLhlMessages.getGoldenLocaleArgumentIds(goldenLhl);
+  return cullObsoleteLhlMessages.cullLocale(goldenLocaleArgumentIds, localeLhl, loggedCulls);
+}
+
+describe('Cull obsolete translations', () => {
+  const goldenLhl = {
+    plainString: {
+      message: 'Plain string',
+    },
+    displayValue: {
+      message: 'Total size was {totalBytes, number, bytes} KB',
+    },
+    nestedInPlural: {
+      // Ensure nested arguments are parsed and handled.
+      // eslint-disable-next-line max-len
+      message: '{requestCount, plural, =1 {1 request • {byteCount, number, bytes} KB} other {# requests • {byteCount, number, bytes} KB}}',
+    },
+  };
+
+  it('removes nothing if the messages are exactly the same', () => {
+    const culledLocale = cullLocale(goldenLhl, goldenLhl);
+    expect(culledLocale).toEqual(goldenLhl);
```

Comment:
nit: could we clone one of them to make sure it's not mutating goldenLhl or relying on referential equality or something? :)